parts/part.go

```diff
@@ -334,6 +335,10 @@ func FindMaximumNonOverlappingSet(schema *dynparquet.Schema, parts []*Part) ([]*
 			// The current part must be removed. Don't update prevEnd or prev,
 			// this will be used in the next iteration and must stay the same.
 			overlapping = append(overlapping, parts[i])
+
+			if i == len(parts)-1 { // This is the last iteration mark this one as missing
+				missing = parts[prev]
```

Comment:
instead of adding a `missing` variable, why not just `overlapping = append(overlapping, parts[prev])` here?

Reply:
I found that sometimes conflicted with the check below and caused the part to be double saved in some cases